src/ucp/tag/tag_match.c

```diff
@@ -85,6 +85,21 @@ int ucp_tag_exp_remove(ucp_tag_match_t *tm, ucp_request_t *req)
     return 0;
 }
 
+void ucp_tag_exp_erase(ucp_tag_match_t *tm, ucp_tag_t tag)
+{
+    ucp_request_queue_t *req_queue = ucp_tag_exp_get_queue_for_tag(tm, tag);
```

Comment:
there may be a tag collision, right? would need to compare tags for exact match.